### PR TITLE
Problem: Travis does not test building docs

### PR DESCRIPTION
**Pull request by dkliban**
Solution: enable docs testing template_config.yaml

This patch adds the latest plugin template changes in addition to the documentation
building on Travis.

The latest changes make the check for services being started shorter. The logs for
pulp-api are also longer in case of test failures.

[noissue]

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -13,3 +13,39 @@ Changelog
 
 .. towncrier release notes start
 
+2.0.0a2.dev1 (2019-09-06)
```
Comment (mdellweg) [inline CHANGES.rst]: That should be the new version, i think.

File: CHANGES.rst [hunk 2]
```diff
@@ -13,3 +13,39 @@ Changelog
...
+- Reference the fact you must have both _relative_path and relative_path.
+  `#5376 <https://pulp.plan.io/issues/5376>`_
+- Fix various documentation issues from API changes, plus other misc fixes.
+  `#5380 <https://pulp.plan.io/issues/5380>`_
+
+
+Misc
+----
+
+- `#4681 <https://pulp.plan.io/issues/4681>`_, `#5401 <https://pulp.plan.io/issues/5401>`_
```
Comment (mdellweg) [inline CHANGES.rst]: I think, the misc entries are important enough to be spelled out. We probably need some configuration for this.

See https://github.com/hawkowl/towncrier/blob/master/pyproject.toml#L7

File: docs/_templates/restapi.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>REST API for Pulp 3 RPM Plugin</title>
```
Comment (mdellweg) [inline docs/_templates/restapi.html]:
```suggestion
    <title>REST API for Pulp 3 Deb Plugin</title>
```

File: docs/Makefile [hunk 4]
```diff
@@ -42,6 +42,7 @@ clean:
 	-rm -rf $(BUILDDIR)/*
 
 html:
+	curl -o _static/api.json "http://localhost:24817/pulp/api/v3/docs/api.json?plugin=pulp_rpm"
```
Comment (mdellweg) [inline docs/Makefile]:
```suggestion
	curl -o _static/api.json "http://localhost:24817/pulp/api/v3/docs/api.json?plugin=pulp_deb"
```

File: CHANGES.rst (hunk 2 above)
Comment (mdellweg) [reply CHANGES.rst]: We could also change the template and ignore the `showcontent` parameter.

Comment (dkliban) [reply CHANGES.rst]: I've added both of these changes to my branch.

File: CHANGES.rst (hunk 1 above)
Comment (dkliban) [reply CHANGES.rst]: I've rebased and re-generated the change log.

File: docs/_templates/restapi.html (hunk 3 above)
Comment (dkliban) [reply docs/_templates/restapi.html]: Thank you!

**Pull request merged**